examples/contracts/generic_contract/src/contract.rs

```diff
@@ -0,0 +1,99 @@
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{Reply, Response, StdResult};
+use serde::de::DeserializeOwned;
+use serde::Deserialize;
+use sylvia::types::{CustomMsg, ExecCtx, InstantiateCtx, MigrateCtx, QueryCtx, ReplyCtx};
+use sylvia::{contract, schemars};
+
+#[cw_serde]
+pub struct ExternalMsg;
+impl cosmwasm_std::CustomMsg for ExternalMsg {}
+
+pub struct GenericContract<InstantiateParam, ExecParam, QueryParam, MigrateParam, RetType>(
+    std::marker::PhantomData<(
+        InstantiateParam,
+        ExecParam,
+        QueryParam,
+        MigrateParam,
+        RetType,
+    )>,
+);
+
+#[contract]
+impl<InstantiateParam, ExecParam, QueryParam, MigrateParam, RetType>
+    GenericContract<InstantiateParam, ExecParam, QueryParam, MigrateParam, RetType>
+where
+    for<'msg_de> InstantiateParam: CustomMsg + Deserialize<'msg_de> + 'msg_de,
+    for<'exec> ExecParam: CustomMsg + DeserializeOwned + 'exec,
+    for<'exec> QueryParam: CustomMsg + DeserializeOwned + 'exec,
+    for<'exec> MigrateParam: CustomMsg + DeserializeOwned + 'exec,
+    for<'ret> RetType: CustomMsg + DeserializeOwned + 'ret,
```

Comment:
I am wondering - shouldn't we go with `+ 'static` instead of a generalized lifetime? Don't really see when you want to use non-static type here.

Reply:
Not sure but I decided to go with custom lifetime to cover expanded predicate. I can change three of them to static